internal/provider/cluster_resource_test.go

```diff
@@ -698,11 +698,12 @@ func TestIntegrationDedicatedClusterResource(t *testing.T) {
 		})
 
 	s.EXPECT().GetCluster(gomock.Any(), clusterID).
-		Return(&firstUpdateCluster, httpOk, nil).Times(7)
+		Return(&firstUpdateCluster, httpOk, nil).Times(6)
```

Comment:
Do you mind separating this fix out into a separate commit.  Same PR is fine.  We should document pretty heavily whats going on here in the commit message.

Reply:
done! Please let me know if anything is missing or if something is incorrect